go/vt/vttablet/tabletserver/vstreamer/main_flaky_test.go

```diff
@@ -38,7 +38,7 @@ var (
 )
 
 func TestMain(m *testing.M) {
-	flag.Parse() // Do not remove this comment, import into google3 depends on it
+	pflag.Parse()
 	ignoreKeyspaceShardInFieldAndRowEvents = true
 	if testing.Short() {
```

Comment:
@ajm188,  should we just be calling `flag.Parse()` here rather than `pflag.Parse()` so that `testing.Short()` works or is there another way to handle this issue?

```
goroutine 1 [running]:
testing.Short(...)
	/opt/hostedtoolcache/go/1.18.4/x64/src/testing/testing.go:534
vitess.io/vitess/go/vt/vttablet/tabletserver/vstreamer.TestMain(0x442831?)
	/home/runner/work/vitess/vitess/go/vt/vttablet/tabletserver/vstreamer/main_flaky_test.go:43 +0xc7
main.main()
	_testmain.go:159 +0x1d3
FAIL	vitess.io/vitess/go/vt/vttablet/tabletserver/vstreamer	0.021s
FAIL
FAILED (try 2/3) in vitess.io/vitess/go/vt/vttablet/tabletserver/vstreamer (return code 1). See above for errors.
panic: testing: Short called before Parse
```

Reply:
You still need to parse on `flag` (because the `testing` package is super inflexible).

For extra safety, I would suggest doing something like:

```go
var tmp []string
os.Args, tmp = os.Args[0:1], os.Args[1:]
flag.Parse()
os.Args = append(os.Args, tmp...)

// rest of the code
```

Reply:
alternatively (my preference, but also more work) we could rewrite this to not depend on `testing.Short`. up to you!

Reply:
> alternatively (my preference, but also more work) we could rewrite this to not depend on `testing.Short`. up to you!

What alternative way are you thinking of here?

Reply:
> For extra safety, I would suggest doing something like:
 
 ```go
 var tmp []string
 os.Args, tmp = os.Args[0:1], os.Args[1:]
 flag.Parse()
 os.Args = append(os.Args, tmp...)
 
> // rest of the code
 ```

What is this code protecting against?

Reply:
> What alternative way are you thinking of here?

not sure, it depends entirely on what the goal of the `Short` check was for here, so if you know that'd be useful to know! for example we could do something like `if os.Getenv("SOME_VARIABLE")` instead but that might not make sense for the use case, or maybe `testing.Short()` is _never_ true in our usage and we should just delete it. it's not really possible for me to say

Reply:
> What is this code protecting against?

`os.Args` is literally just the stuff passed to the program, which in this case is `go test ...`, so if there are other flags in there, this lets us ignore them, because we only care about getting `flag.Parsed()` to return true

Reply:
> > What alternative way are you thinking of here?
> 
> not sure, it depends entirely on what the goal of the `Short` check was for here, so if you know that'd be useful to know! for example we could do something like `if os.Getenv("SOME_VARIABLE")` instead but that might not make sense for the use case, or maybe `testing.Short()` is _never_ true in our usage and we should just delete it. it's not really possible for me to say

For now changed it to flag.Parse() to check if tests pass, which they do. I will bring up the testing.Short() usage in Open Hour. I don't see it getting used in the makefiles or ci workflows anymore, so maybe we can remove that usage entirely.